papers/pydra/paper.rst

```diff
@@ -0,0 +1,965 @@
+
+.. class:: abstract
+
+This paper presents a new lightweight dataflow engine written
+in Python: Pydra. TODO
+
+
+.. class:: keywords
+
+   dataflow engine, scientific workflows, reproducibility
+
+Introduction
+------------
+
+Scientific workflows often require sophisticated analyses that encompass
+a large collection of algorithms.
+The algorithms, that were originally not necessarily designed to work together,
+often written by different authors in different times.
+Some may be written in Python, while others might require calling external programs.
+It is a common practice to create semi-manual workflows that require scientist
+interaction to handle files and partial results between algorithms and external tools.
+This approach is conceptually simple and easy to implement, but the resulting workflow
+is often time consuming, error-prone and difficult to share with others.
+Consistency, reproducibility and scalability demand scientific workflows be organized
+into fully automated pipelines.
+This was the motivation behind writting a new dataflow engine written in Python, *Pydra*,
+that is presented in this paper.
+
+The package is a part of the second generation of the *Nipype* ecosystem
+(:cite:`Gorgolewski2011`, :cite:`Nipype`)
+--- an open-source framework that provides a uniform interface to existing neuroimaging
+software and facilitates interaction between them.
+The *Nipype* project was born in the neuroimaging community, and has been helping scientists build
+workflows for a decade, providing a uniform interface to such neuroimaging packages
+as FSL :cite:`Woolrich09`, ANTs :cite:`ANTs`, AFNI :cite:`AFNI`, FreeSurfer :cite:`FreeSurfer`
+and SPM :cite:`spm_book`.
+This flexibility has made it an ideal basis for popular preprocessing tools,
+such as fMRIPrep :cite:`fMRIPrep` and C-PAC :cite:`C-PAC`.
+The second generation of *Nipype* ecosystem is meant to provide additional flexibility
+and is being developed with reproducibility, ease of use, and scalability in mind.
+*Pydra* itself is a standalone project and is designed as a general-purpose dataflow engine
+to support any scientific domain.
+
+The goal of Pydra is to provide a lightweight dataflow engine for computational graph construction,
+manipulation, and distributed execution, as well as ensuring reproducibility of scientific pipelines.
+In Pydra, a dataflow is represented as a directed acyclic graph, where each node represents a 
+Python function, execution of an external tool, or another reusable dataflow.
+The combination of several key features makes Pydra a customizable and powerful dataflow engine:
+
+* **Composable dataflows:** Any node of a dataflow graph can be another dataflow,
+  allowing for nested dataflows to arbitrary depths and encourages creating reusable dataflows.
+
+* **Flexible semantics for creating nested loops over input sets:**
+  Any *Task* or dataflow can be run over input parameter sets and the outputs can be recombined
```

Comment:
```suggestion
  Any Task or dataflow can be run over input parameter sets and the outputs can be recombined
```

Reply:
Czemu Task z dużej a dataflow z małej?

Reply:
za tym jest pewna mysl, byc moze nie najlpesza, ale jest... chcialam uzywac z duzej tych obiektow bardziej pydrowych. pozniej to jeszcze przemysle